src/ert/field_utils/field_utils.py

```diff
@@ -262,3 +269,220 @@ def save_field(
         export_grdecl(field, output_path, field_name, binary=True)
     else:
         raise ValueError(f"Cannot export, invalid file format: {file_format}")
+
+
+def transform_positions_to_local_field_coordinates(
+    coordsys_origin: tuple[float, float],
+    coordsys_rotation_angle: float,
+    utmx: npt.NDArray[np.float64],
+    utmy: npt.NDArray[np.float64],
+) -> tuple[npt.NDArray[np.float64], npt.NDArray[np.float64]]:
+    """
+    Calculates the coordinates of the input (x,y) positions
+    in local coordinates defined by the local coordinate system.
+    The input coordinates are assumed to be in global (utm) coordinates.
+    The return values are coordinate values in local coordinate system.
+    """
+    # Translate
+    x1 = utmx - coordsys_origin[0]
+    y1 = utmy - coordsys_origin[1]
+    # Rotate
+    # Input angle is the local coordinate systems rotation
+    # anticlockwise relative to global x-axis in degrees
+    rotation_of_ertbox = coordsys_rotation_angle
+    rotation_angle = rotation_of_ertbox * np.pi / 180.0
+    cos_theta = np.cos(rotation_angle)
+    sin_theta = np.sin(rotation_angle)
+    x2 = x1 * cos_theta + y1 * sin_theta
+    y2 = -x1 * sin_theta + y1 * cos_theta
+    return x2, y2
+
+
+def transform_local_ellipse_angle_to_local_coords(
+    coordsys_rotation_angle: float,
+    ellipse_anisotropy_angle: npt.NDArray[np.double],
+) -> npt.NDArray[np.double]:
+    """
+    The input angles for orientation of the localization ellipses
+    are relative to the global (utm) coordinates.
+    The output is the angles relative to the local coordinate system.
+    """
+    # Both angles measured anti-clock from global coordinate systems x-axis in degrees
+    ellipse_anisotropy_transformed = ellipse_anisotropy_angle - coordsys_rotation_angle
+    return ellipse_anisotropy_transformed
+
+
+def localization_scaling_function(
+    distances: npt.NDArray[np.float64],
+    scaling_func: ScalingFunctions = ScalingFunctions.gaspari_cohn,
+) -> npt.NDArray[np.float64]:
+    """
+    Description: Calculate scaling factor to be used as values in
+    RHO matrix in distance-based localization.
+    Take as input array with normalized distances
+    and returns scaling factors.
+
+    :param distances: Array with distances
+    :type distances: npt.NDArray[np.float64]
+    :param scaling_func: Name of the scaling function type
+    :type scaling_func: ScalingFunctions
+    :return: Array with scaling values
+    :rtype: NDArray[float64]
+    """
+    assert isinstance(scaling_func, ScalingFunctions)
+    if scaling_func == ScalingFunctions.gaussian:
```

Comment:
I had a small discussion with @oyvindeide and we might start with a single ScalingFunction (gaspari-cohn).
Or are there currently any reasons why we should be able to have all 3 implementations?

Reply:
I agree. but I don't know which one is best, but gaspari_cohn functions is often used.

Reply:
I will now use only gaspari-cohn function